Sentry won't compile if SDK Tools 31 are not installed

### What React Native libraries do you use?

React Navigation

### Are you using sentry.io or on-premise?

sentry.io (SaS)

### Are you using any other error monitoring solution alongside Sentry?

No

### Other Error Monitoring Solution Name

_No response_

### @sentry/react-native SDK Version

8.9.1

### How does your development environment look like?

````
MacBook Pro M2  v15.7.4 (24G517)
Expo 54 (CNG)
Android SDK 36 + SDK Tools 36+35
````

### Sentry.init()

````js
  Sentry.init({
    integrations: [
      Sentry.breadcrumbsIntegration({ fetch: false, xhr: true }),
      Sentry.graphqlIntegration({ endpoints: [Env.EXPO_PUBLIC_API_URL + '/v2/graphql'] })
    ],
    dsn: Env.EXPO_PUBLIC_SENTRY_KEY,
    environment: Env.EXPO_PUBLIC_SENTRY_ENV,
    enabled: SENTRY_ENABLED_ENVS.includes(Env.EXPO_PUBLIC_NAME),
    sendDefaultPii: true,
    release: `${Application.applicationId}@${Application.nativeApplicationVersion}`,
    dist: Application.nativeBuildVersion || '1',
    appHangTimeoutInterval: 3,
    beforeBreadcrumb: (breadcrumb, _) => {
      if (breadcrumbToIgnore(breadcrumb)) {
        return null;
      }

      return withGraphQLBreadcrumb(withTouchBreadcrumb(breadcrumb));
    },
    denyUrls: [REACHABILITY_URL]
  });
````

### Steps to Reproduce

1. Install only SDK 36 and SDK Tools 36 + 35
2. Update sentry from 7.x to 8.x
3. Try to build with any of available commands (directly gradle or expo ones)

### Expected Result

1. Build completes without errors.
2. `expo-handler` from [@sentry/react-native/android/expo-handler](https://github.com/getsentry/sentry-react-native/blob/9b6b2bacbfb47395b52c9b44ee6384373648356e/packages/core/android/expo-handler/build.gradle#L4) uses the `safeExtGet` to get the values

```
def safeExtGet(prop, fallback) {
    rootProject.ext.has(prop) ? rootProject.ext.get(prop) : fallback
}
```

### Actual Result

```
FAILURE: Build failed with an exception.

* What went wrong:
Could not determine the dependencies of task ':sentry-react-native-expo:generateDebugRFile'.
> Failed to find target with hash string 'android-31' in: /Users/<user>/Library/Android/sdk

Error: /Users/<user>/Developer/mobile-app/android/gradlew app:assembleDebug -x lint -x test --configure-on-demand --build-cache -PreactNativeDevServerPort=8081 -PreactNativeArchitectures=arm64-v8a exited with non-zero code: 1
```

### Proposed fix
It's fixed if setting that value to 36 instead of 31. So probably using `safeExtGet` in that subproject would be enough.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -15,6 +15,7 @@
 
 ### Fixes
 
+- Fix Android build failure in `expo-handler` when Android SDK 31 is not installed by using `safeExtGet` for `compileSdkVersion` and `minSdkVersion` ([#6061](https://github.com/getsentry/sentry-react-native/pull/6061))
 - Stop the Hermes sampling profiler on React instance teardown to prevent `pthread_kill` SIGABRT when the JS thread is torn down with profiling active ([#6035](https://github.com/getsentry/sentry-react-native/pull/6035))
 - Mask the Sentry auth token in the `sentry.gradle` upload-task lifecycle log ([#6057](https://github.com/getsentry/sentry-react-native/pull/6057))
 - Discard invalid navigation/interaction transactions via an event processor instead of mutating the internal `_sampled` flag, removing misleading "dropped due to sampling" debug logs ([#6051](https://github.com/getsentry/sentry-react-native/pull/6051))

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -15,7 +15,7 @@
 
 ### Fixes
 
-- Fix Android build failure in `expo-handler` when Android SDK 31 is not installed by using `safeExtGet` for `compileSdkVersion` and `minSdkVersion` ([#6061](https://github.com/getsentry/sentry-react-native/pull/6061))
+- Android build failure in `expo-handler` when Android SDK 31 is not installed by using `safeExtGet` for `compileSdkVersion` and `minSdkVersion` ([#6061](https://github.com/getsentry/sentry-react-native/pull/6061))
 - Stop the Hermes sampling profiler on React instance teardown to prevent `pthread_kill` SIGABRT when the JS thread is torn down with profiling active ([#6035](https://github.com/getsentry/sentry-react-native/pull/6035))
 - Mask the Sentry auth token in the `sentry.gradle` upload-task lifecycle log ([#6057](https://github.com/getsentry/sentry-react-native/pull/6057))
 - Discard invalid navigation/interaction transactions via an event processor instead of mutating the internal `_sampled` flag, removing misleading "dropped due to sampling" debug logs ([#6051](https://github.com/getsentry/sentry-react-native/pull/6051))

--- a/packages/core/android/expo-handler/build.gradle
+++ b/packages/core/android/expo-handler/build.gradle
@@ -1,11 +1,15 @@
+def safeExtGet(prop, fallback) {
+    rootProject.ext.has(prop) ? rootProject.ext.get(prop) : fallback
+}
+
 apply plugin: 'com.android.library'
 
 android {
-    compileSdkVersion 31
+    compileSdkVersion safeExtGet('compileSdkVersion', 31)
     namespace = "io.sentry.react.expo"
 
     defaultConfig {
-        minSdkVersion 21
+        minSdkVersion safeExtGet('minSdkVersion', 21)
     }
 
     compileOptions {

